addons/mrp/report/mrp_report_mo_overview.py

```diff
@@ -751,23 +759,38 @@ def _add_origins_to_forecast(self, forecast_lines):
                     # Remove 'in_transit' for MTO replenishments
                     line['in_transit'] = False
                     move_origin_qty = move_origin.product_uom._compute_quantity(move_origin.product_uom_qty, line['uom_id'])
+                    if not origin_qty.get(doc_origin):
+                        origin_qty[doc_origin] = move_origin_qty
                     # Move quantity matches forecast, can add origin to the line
-                    if float_compare(line['quantity'], move_origin_qty, precision_rounding=line['uom_id'].rounding) == 0:
+                    if float_compare(line['quantity'], origin_qty[doc_origin], precision_rounding=line['uom_id'].rounding) == 0:
+                        # if the doc_origin is already displayed, this line is a duplicate
+                        if doc_origin.id in displayed_origin_ids_per_model[doc_origin._name]:
+                            forecast_lines.pop(forecast_lines.index(line))
                         line['document_in'] = {'_name': doc_origin._name, 'id': doc_origin.id}
                         line['move_in'] = move_origin
+                        origin_qty[doc_origin] -= 0
+                        displayed_origin_ids_per_model[doc_origin._name].append(doc_origin.id)
                         break
 
                     # Quantity doesn't match, either multiple origins for a single line or multiple lines for a single origin
-                    used_quantity = min(move_out_qty, move_origin_qty)
-                    new_line = copy.copy(line)
-                    new_line['quantity'] = used_quantity
-                    new_line['document_in'] = {'_name': doc_origin._name, 'id': doc_origin.id}
-                    new_line['move_in'] = move_origin
-                    new_lines.append(new_line)
+                    used_quantity = min(move_out_qty, origin_qty[doc_origin])
                     # Remove used quantity from original forecast line
                     line['quantity'] -= used_quantity
-
+                    origin_qty[doc_origin] -= used_quantity
                     move_out_qty -= used_quantity
+
+                    if not doc_origin.id in displayed_origin_ids_per_model[doc_origin._name]:
+                        new_line = copy.copy(line)
+                        new_line['quantity'] = used_quantity
+                        new_line['document_in'] = {'_name': doc_origin._name, 'id': doc_origin.id}
+                        new_line['move_in'] = move_origin
+                        new_lines.append(new_line)
+
+                        # line is fully dispatched
+                        # prevent future lines to display a 0 quantity
+                        if float_compare(origin_qty[doc_origin], 0, line['move_out'].product_uom.rounding) <= 0:
```

Comment:
No guarantee `origin_qty[doc_origin]` is the same `uom` as `line['move_out']`, especially since it's supposed to come from `line['move_in']`